File: samples/CSharp/MongoDB.SingleReplica/AppHost.cs
#pragma warning disable ASPIRECERTIFICATES001

using Fedarovich.Aspire.Hosting.MongoDB.ReplicaSet;
using Projects;
using Shared;

var builder = DistributedApplication.CreateBuilder(args);

var mongoServer = builder.AddMongoDB("TestMongo", 27017);

mongoServer.AddDatabase("TestMongoDB", "TestDB");

var mongoRs = builder.AddMongoDBReplicaSet("TestMongoRS")
    .WithMember(mongoServer)
    .WithDbGate();

// Add sample services that use the replica set.

var dotNetLocal = builder.AddProject<WebApi>("DotNet-Local")
    .WithReference(mongoRs);

var dotNetContainer = builder.AddDockerfileForProject<WebApi>("DotNet-Container")
    .WithReference(mongoRs);

var nodeLocal = builder.AddJavaScriptApp("NodeJS-Local", "../../TypeScript/WebApi")
    .WithHttpEndpoint(targetPort: 3000)
    .WithReference(mongoRs);

var nodeContainer = builder.AddDockerfile("NodeJS-Container", "../../TypeScript/WebApi")
    .WithHttpEndpoint(targetPort: 3000)
    .WithReference(mongoRs);

var pythonContainer = builder.AddDockerfile("Python-Container", "../../Python/WebApi")
    .WithHttpEndpoint(targetPort: 8000)
    .WithReference(mongoRs);

builder.Build().Run();

File: samples/TypeScript/MongoDB.SingleReplica/apphost.ts
// Aspire TypeScript AppHost
// For more information, see: https://aspire.dev

import { createBuilder } from './.modules/aspire.js';

const builder = await createBuilder();

const mongoServer = await builder.addMongoDB('TestMongo', { port: 27017 });

await mongoServer.addDatabase('TestMongoDB');

const mongoRS = await builder.addMongoDBReplicaSet('TestMongoRS')
    .withMember(mongoServer)
    .withDbGate();

// Add sample services that use the replica set.

const dotNetLocal = await builder.addProject('DotNet-Local', '../../CSharp/WebApi/WebApi.csproj', 'Container (Dockerfile)')
    .withHttpEndpoint({ targetPort: 8080 })
    .withHttpsEndpoint({ targetPort: 8081 })
    .withReference(mongoRS);

const dotNetContainer = await builder.addDockerfileForProject('DotNet-Container', '../../CSharp/WebApi/WebApi.csproj')
    .withReference(mongoRS);

const nodeLocal = await builder.addJavaScriptApp('NodeJS-Local', '../WebApi')
    .withHttpEndpoint({ targetPort: 3000 })
    .withReference(mongoRS);

const nodeContainer = await builder.addDockerfile('NodeJS-Container', '../WebApi')
    .withHttpEndpoint({ targetPort: 3000 })
    .withReference(mongoRS);

const pythonContainer = await builder.addDockerfile('Python-Container', '../../Python/WebApi')
    .withHttpEndpoint({ targetPort: 8000 })
    .withReference(mongoRS);

const goContainer = await builder.addDockerfile('Go-Container', '../../Go/WebApi')
    .withHttpEndpoint({ targetPort: 8080 })
    .withReference(mongoRS);

await builder.build().run();